R/BASiCS_DivideAndConquer.R

```diff
@@ -0,0 +1,115 @@
+#' Run divide and conquer MCMC with BASiCS
+#' 
+#' Performs MCMC inference on batches of data. \code{Data} is divided
+#' into \code{NSubsets} batches, and \code{BASiCS_MCMC} is run on each
+#' batch separately.
+#' 
+#' Subsets are chosen such that the average library size (when partitioning
+#' by cells) or average count (when partitioning by genes) is not significantly
+#' different between batches, at a significance level \code{Alpha}.
+#' 
+#' @param Data SingleCellExperiemnt object
+#' @param NSubsets The number of batches to create and perform MCMC inference
+#' with.
+#' @param SubsetBy A character value specifying whether batches should consist
+#' of a subset of the cells in \code{Data} (when \code{SubsetBy="cell"})
+#' or a subset of the genes in \code{Data} (when \code{SubsetBy="gene"}).
+#' @param Alpha A numeric value specifying the statistical significance level
+#' used to determine whether the average library size or average count
+#' are significantly different between batches.
+#' @param WithSpikes,Regression,PriorParam See \code{\link{BASiCS_MCMC}}.
+#' @param BPPARAM A \code{\link{BiocParallelParam}} instance.
+#' @param ... Passed to  \code{\link{BASiCS_MCMC}}.
+#' @return A list of \linkS4class{BASiCS_Chain} objects.
+#' @examples
+#'  bp <- BiocParallel::SnowParam()
+#'  Data <- BASiCS_MockSCE()
+#'  BASiCS_DivideAndConquer(
+#'    Data, 
+#'    NSubsets = 2,
+#'    SubsetBy = "gene",
+#'    N = 8,
+#'    Thin = 2,
+#'    Burn = 4,
+#'    WithSpikes = TRUE,
+#'    Regression = TRUE,
+#'    BPPARAM = bp
+#'  )
+#' @references
+#' Simple, Scalable and Accurate Posterior Interval Estimation
+#' Cheng Li and Sanvesh Srivastava and David B. Dunson
+#' arXiv (2016)
+#' @export
+BASiCS_DivideAndConquer <- function(
```

Comment:
I also suggest to have the parameters in the same order as in `BASiCS_MCMC`

Reply:
There's no need to know or check what the values of thin, burn, etc are at this level, they'd be passed into _MCMC anyway

Reply:
OK. However, as these parameters are a requirement for `BASiCS_MCMC`, we should describe this in the documentation (`BASiCS_DivideAndConquer` will fail via `BASiCS_MCMC` otherwise). Perhaps you can add a note to the line below, stating that all the parameters that are required for `BASiCS_MCMC` should be passed to this function too?

```
@param ... Passed to  \code{\link{BASiCS_MCMC}}.
```